projects/core/src/main/java/dan200/computercraft/core/filesystem/FileMount.java

```diff
@@ -143,6 +143,14 @@ public boolean isDirectory(String path) {
         return file.exists() && file.isDirectory();
     }
 
+    @Override
+    public boolean isReadOnly(String path) throws IOException {
+        if (!created()) return path.isEmpty();
+
+        var file = getRealPath(path);
+        return file.exists() && !file.canWrite();
```

Comment:
I'm not sure about this. Just to confirm my understanding, if we've not created the mount, then `/` is considered read-only? However, this should only be the case if the actual "computercraft/computers" directory is read only, which hopefully isn't the case!

I wonder how we should handle missing files in general here. Currently `fs.isReadOnly("read-only-dir/missing-file")` will return `false`, which isn't great. A couple of possibilities I can think of:

 - Keep the behaviour here: it's not a big issue after all!

 - `isReadOnly` walks up the tree until it finds a file which exists, then returns that. My big concern here is someone doing `fs.isReadOnly(("/p"):rep(10000))`, which could cause us to spend ages in filesystem code. 

   Hrmr, maybe you could do a binary search over the file path looking for one which exists :thinking:. Feel like that might be a bit too complicated.

Reply:
Looks like that was a total oversight with the created check there. I've removed the check entirely and now I believe the behaviour is correct. I went with traversing the tree in the same way `makeDirectory` does. In practice, 10000 repetitions is near-instantaneous, and 100000 repetitions will throw "Too long without yielding". Since `makeDirectory` would also have the same issue, I don't think this is a major concern for now.

![javaw_Y0HY8Bis14](https://user-images.githubusercontent.com/858456/203847247-0b1def62-a562-4db0-8376-3193dca12648.png)